fix(vercel): validate remote domains

farnabaz opened the pull request:
Add logic to serve original src if image does not serve from listed domains

Resolves nuxt/image#342

pi0 commented:
Thanks for pull-request @farnabaz. I think we shall move this validation out of provider implementation and always validate to externalize URLs when `domains` option is provided and current URL is not matching.

Atinux commented:
On which file/function can he implanted the validation @pi0

pi0 commented:
I think here we can add the hostname check:

https://github.com/nuxt/image/blob/d9aeae2144ac815f45081d06f1b6b60e62e2ec14/src/runtime/image.ts#L98

farnabaz commented:
I moved domain validation into `resolveImage` helper and removed logic from both `ipx` and `vercel` providers.

Does it requires other changes @pi0?

pi0 commented:
Changes look good @farnabaz 👌 I'm adding a change to ipx using hostname to use unified method and avoid startsWith check for vercel

pi0 commented:
- Updated `ipx` with hostname check
- Domains are normalized in module
- Only do external check if provider exports `validateDomains` flag enabled for `ipx` and `vercel`. This is to avoid breaking `prismic` and `storyblok` as they accept full URLs (will probably add a flag for this incompatible providers instead later for now flag is undocumented)
